internal/manifests/collector/annotations.go

```diff
@@ -27,8 +27,25 @@ func Annotations(instance v1beta1.OpenTelemetryCollector) (map[string]string, er
 	// new map every time, so that we don't touch the instance's annotations
 	annotations := map[string]string{}
 
-	// Enable Prometheus annotations by default if DisablePrometheusAnnotations is nil or true
-	if !instance.Spec.Observability.Metrics.DisablePrometheusAnnotations {
+	var enableAnnotations bool
+	// enableAnnotations is true only if both disable variables are false.
+	// DisableAutomaticPrometheusAnnotations takes precedence over DisablePrometheusAnnotations.
+	if instance.Spec.Observability.Metrics.DisableAutomaticPrometheusAnnotations {
```

Comment:
you'll need to add this to the v1beta1 type too

Reply:
looks fixed in beta version